source/common/formatter/substitution_formatter.cc

```diff
@@ -1918,6 +1926,67 @@ UpstreamHostMetadataFormatter::UpstreamHostMetadataFormatter(const std::string&
                           return host->metadata().get();
                         }) {}
 
+ProxyProtocolTlvsFormatter::ProxyProtocolTlvsFormatter(const std::string& tlv_type_str) {
+  // Specified tlv_type must be parsable as an int.
+  TRY_ASSERT_MAIN_THREAD {
+    tlv_type_ = std::stoi(tlv_type_str);
+  } 
+  END_TRY
+  catch (const std::exception& ex) {
+    throw EnvoyException(fmt::format(
+        "Invalid parameter provided for PROXY_PROTOCOL_TLVS header: {}. Not parsable as int.",
+        tlv_type_str));
+  }
+  // Check if a valid TLV type was passed in.
+  if (tlv_type_ >= 256 || tlv_type_ <= 0) {
+    throw EnvoyException(fmt::format("Invalid parameter provided for PROXY_PROTOCOL_TLVS header: "
+                                     "{}. Must be a positive integer less than 256.",
+                                     tlv_type_str));
+  }
+}
+
+// Parses the proxy protocol TLVs stored in the FilterState with the key
+// "envoy.network.proxy_protocol_options" and returns a function suitable for accessing the
```

Comment:
I don't understand how this "returns a function". I feel the purpose of this function is clear since it's just to implement the interface. The implementation is also very clear, so we don't have to explain how it's implemented.

Reply:
It doesn't return a function 🤔 I don't remember why I wrote this particular comment, but I just deleted it in my last commit. Thanks for the callout!